lib/spack/spack/cmd/testinstall.py

```diff
@@ -0,0 +1,181 @@
+    
+    #always true for test
+    #subparser.add_argument(
+    #    '--keep-stage', action='store_true', dest='keep_stage',
+    #    help="Don't remove the build stage if installation succeeds.")
+    
+    subparser.add_argument(
+        '-n', '--no-checksum', action='store_true', dest='no_checksum',
+        help="Do not check packages against checksum")
+    subparser.add_argument(
+        '-v', '--verbose', action='store_true', dest='verbose',
+        help="Display verbose build output while installing.")
+    
+    #subparser.add_argument(
+    #    '--fake', action='store_true', dest='fake',
+    #    help="Fake install.  Just remove the prefix and touch a fake file in it.")
+    
+    subparser.add_argument(
+        'output', help="test output goes in this file")
+    
+    subparser.add_argument(
+        'packages', nargs=argparse.REMAINDER, help="specs of packages to install")
+
+
+class JunitResultFormat(object):
+    def __init__(self):
+        self.root = ET.Element('testsuite')
+        self.tests = []
+        
+    def addTest(self, buildId, passed=True, buildInfo=None):
+        self.tests.append((buildId, passed, buildInfo))
+    
+    def writeTo(self, stream):
+        self.root.set('tests', '{0}'.format(len(self.tests)))
+        for buildId, passed, buildInfo in self.tests:
+            testcase = ET.SubElement(self.root, 'testcase')
+            testcase.set('classname', buildId.name)
+            testcase.set('name', buildId.stringId())
+            if not passed:
+                failure = ET.SubElement(testcase, 'failure')
+                failure.set('type', "Build Error")
+                failure.text = buildInfo
+        ET.ElementTree(self.root).write(stream)
+
+
+class BuildId(object):
+    def __init__(self, name, version, hashId):
+        self.name = name
+        self.version = version
+        self.hashId = hashId
+    
+    def stringId(self):
+        return "-".join(str(x) for x in (self.name, self.version, self.hashId))
+
+
+def createTestOutput(spec, handled, output):
+    if spec in handled:
+        return handled[spec]
+    
+    childSuccesses = list(createTestOutput(dep, handled, output) 
+            for dep in spec.dependencies.itervalues())
+    package = spack.db.get(spec)
+    handled[spec] = package.installed
+    
+    if all(childSuccesses):
+        bId = BuildId(spec.name, spec.version, spec.dag_hash())
+
+        if package.installed:
+            buildLogPath = spack.install_layout.build_log_path(spec)
+        else:
+            #TODO: search recursively under stage.path instead of only within
+            #    stage.source_path
+            buildLogPath = join_path(package.stage.source_path, 'spack-build.out')            
+
+        with open(buildLogPath, 'rb') as F:
+            buildLog = F.read() #TODO: this may not return all output
+            #TODO: add the whole build log? it could be several thousand
+            #    lines. It may be better to look for errors.
```

Comment:
Every build in bamboo has a build log ([see here](https://confluence.atlassian.com/bamboo/viewing-a-build-result-289276921.html)).  So it's probably not necessary to duplicate the entire output per test case.  I think the error message should probably be there, though.

Can you search for the first occurrence of `error` (case insensitive) in the output and maybe print from that to the end of the build output?  If there's no `error` or if it would be too long to do it this way you could store some fixed number of lines of the tail of the output of the test.